proposals/0000-exportable-named-default.rst

```diff
@@ -0,0 +1,466 @@
+described in the next subsection.
+
+Note that a ``default`` declaration that repeats a type name more than once is perfectly valid, and sometimes may
+be necessary to resolve coflicts. For example, a module that imports two conflicting defaults
+
+::
+
+   default C (Int, Bool)
+
+and
+   
+::
+
+   default C (Bool, Int)
+
+may use a local declaration
+
+::
+
+   default C (Int, Bool, Int)
+
+to override the imports. Because this declaration subsumes both imported defaults it will not trigger any compiler
+warning. When used to resolve ambiguity (next section) it behaves exactly like ``default C( Int, Bool)``; that is, the
+repeats can be discarded.
+
+   
+Rules for disambiguation at the use site
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The disambiguation rules are a conservative extension of the existing rules in Haskell 2010, which state that
+ambiguous type variable *v* is defaultable if:
+
+    - *v* appears only in constraints of the form *C* *v*, where *C* is a class, and
+
+    - at least one of these classes is a numeric class, (that is, ``Num`` or a subclass of ``Num``), and
+
+    - all of these classes are defined in the Prelude or a standard library.
+
+    Each defaultable variable is replaced by the first type in the default list that is an instance of all the
+    ambiguous variable’s classes. It is a static error if no such type is found.
+
+The new rules instead require only that 
+
+- *v* appears in at least one constraint of the form *C* *v*, where *C* is a single-parameter class.
+
+Informally speaking, the type selected for defaulting is the first type from the ``default`` list for class *C* that
+satisfies all constraints on type variable *v*. If there are multiple *C*\ `i`:subscript: *v* constraints with
+competing ``default`` declarations, they have to resolve to the same type.
+
+To make the design more explicit, the following algorithm *can* be used for default resolution, but any other method
+that achieves the same effect can be substitued:
+
+Let *S* be the complete set of unsolved constraints, and initialize *S*\ `x`:subscript: to an empty set of constraints.
+For every *v* that is free in *S*:
+
+1. Define *C*\ `v`:subscript: = { *C*\ `i`:subscript: v | *C*\ `i`:subscript: v ∈ *B*\ `v`:subscript: }, the subset of
+   *S* consisting of all constraints in *S* of form (*C*\ `i`:subscript: v), where *C*\ `i`:subscript: is a
+   single-parameter type class.
+2. Define *D*\ `v`:subscript:, by extending *C*\ `v`:subscript: with the superclasses of every *C*\ `i`:subscript: in
+   *C*\ `v`:subscript:
+3. Define *E*\ `v`:subscript:, by filtering *D*\ `v`:subscript: to contain only classes with a default declaration.
+4. For each *C*\ `i`:subscript: in *E*\ `v`:subscript:, find the first type *T*\ `i`:subscript: in the default list
+   for *C*\ `i`:subscript: that satisfies all required constraints from the set *C*\ `v`:subscript:.
```

Comment:
It'd be more precise to say "...for which, for every (Ci v) in Cv, the constraint (Ci T) is soluble".

(To avoid suffix confusion, I'd remove the "i" subscript from T.